### PR TITLE
[SPARK-25445][BUILD][FOLLOWUP] Resolve issues in release-build.sh for publishing scala-2.12 build

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

This is a follow up for #22441.

1. Remove flag "-Pkafka-0-8" for Scala 2.12 build.
2. Clean up the script, simpler logic.
3. Switch to Scala version to 2.11 before script exit.

## How was this patch tested?

Manual test.

### CODE REVIEW

**Comment by gengliangwang** [comment]
@cloud-fan @srowen

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -305,16 +290,17 @@ if [[ "$1" == "package" ]]; then
...
 
   if [[ $PUBLISH_SCALA_2_12 = 1 ]]; then
+    ./dev/change-scala-version.sh 2.12
     key="without-hadoop-scala-2.12"
     args="-Phadoop-provided"
     extra=""
-    if ! make_binary_release "$key" "2.12" "$args" "$extra"; then
+    if ! make_binary_release "$key" "$SCALA_2_12_PROFILES $args" "$extra"; then
       error "Failed to build $key package. Check logs for details."
     fi
```
Comment (cloud-fan) [inline dev/create-release/release-build.sh]: shall we change the scala version back to 2.11 here?

File: dev/create-release/release-build.sh [hunk 2]
```diff
@@ -446,6 +432,8 @@ if [[ "$1" == "publish-release" ]]; then
   # Clean-up Zinc nailgun process
   $LSOF -P |grep $ZINC_PORT | grep LISTEN | awk '{ print $2; }' | xargs kill
 
+  ./dev/change-scala-version.sh 2.11
```
Comment (cloud-fan) [inline dev/create-release/release-build.sh]: To make the code cleaner, I'd suggest we change the scala version back right after we finish building scala 2.10 or 2.12, e.g.
```
... build with scala 2.11

if PUBLISH_SCALA_2_10 = 1
  ./dev/change-scala-version.sh 2.10
  ... build with scala 2.10
  ./dev/change-scala-version.sh 2.11

if PUBLISH_SCALA_2_12 = 1
  ./dev/change-scala-version.sh 2.12
  ... build with scala 2.12
  ./dev/change-scala-version.sh 2.11
```

cc @srowen

**Comment by cloud-fan** [comment]
thanks, the code looks much simpler! I'll test it out tomorrow

File: dev/create-release/release-build.sh (hunk 2 above)
Comment (srowen) [reply dev/create-release/release-build.sh]: I guess it couldn't hurt to carefully change it back right after each step is done that needs a different version. As long as it probably happens in the end no matter what.

I am fine with the other changes.

**Comment by cloud-fan** [comment]
I've tested it manually to create a scala 2.12 build and also a scala 2.12 maven package. It works well, I'm merging it to master/2.4, thanks!

**Pull request closed without merging**